target/bin/setquota

```diff
@@ -10,43 +10,44 @@ source /usr/local/bin/helpers/index.sh
 
 DATABASE=${DATABASE:-/tmp/docker-mailserver/dovecot-quotas.cf}
 USER_DATABASE=${USER_DATABASE:-/tmp/docker-mailserver/postfix-accounts.cf}
+MESSAGE_INVALID_FORMAT='Invalid quota format. e.g. 302M (B (byte), k (kilobyte), M (megabyte), G (gigabyte) or T (terabyte))'
 
 USER="${1}"
 shift
 QUOTA="${*}"
 
-function usage  { echo "Usage: setquota <user@domain> [<quota>]" ; }
+function usage { echo 'Usage: setquota <user@domain> [<quota>]' ; }
 
-[[ -z ${USER} ]] && { usage ; _exit_with_error "no username specified" ; }
-[[ ${USER} =~ .*\@.* ]] || { usage ; _exit_with_error "username must include the domain" ; }
+[[ -z ${USER} ]] && { usage ; _exit_with_error "No username specified" ; }
+[[ ${USER} =~ .*\@.* ]] || { usage ; _exit_with_error "Username must include the domain" ; }
 
 if ! grep -qE "^${USER}\|" "${USER_DATABASE}"
 then
-  usage; _exit_with_error "user ${USER} does not exist"
+  usage; _exit_with_error "User '${USER}' does not exist"
 fi
 
 # check quota
 if [[ -n ${QUOTA} ]] && ! echo "${QUOTA}" | grep -qE "^([0-9]+(B|k|M|G|T)|0)\$"
 then
   usage
-  _exit_with_error "invalid quota format. e.g. 302M (B (byte), k (kilobyte), M (megabyte), G (gigabyte) or T (terabyte))"
+  _exit_with_error "${MESSAGE_INVALID_FORMAT}"
 fi
 
 _create_lock # Protect config file with lock to avoid race conditions
 
 touch "${DATABASE}"
 if [[ -z ${QUOTA} ]]
 then
-  read -r -s "Enter quota (e.g. 10M): " QUOTA
+  read -r -s 'Enter quota (e.g. 10M): ' QUOTA
   echo
-  [[ -z "${QUOTA}" ]] && _exit_with_error "Quota must not be empty. Use 0 for unlimited quota"
+  [[ -z "${QUOTA}" ]] && _exit_with_error 'Quota must not be empty (use 0 for unlimited quota)'
 fi
 
 # check quota
 if [[ -n ${QUOTA} ]] && ! echo "${QUOTA}" | grep -qE "^([0-9]+(B|k|M|G|T)|0)\$"
 then
   usage
-  _exit_with_error "invalid quota format. e.g. 302M (B (byte), k (kilobyte), M (megabyte), G (gigabyte) or T (terabyte))"
+  _exit_with_error "${MESSAGE_INVALID_FORMAT}"
 fi
```

Comment:
This doesn't address the concern fully ([_originally raised in this review comment_](https://github.com/docker-mailserver/docker-mailserver/pull/2500#discussion_r831638059)), but is an improvement at least.

I'm still curious why the entire conditional statement is repeated, seems unnecessary?

Reply:
It actually is :D Addressed in 9d3eda4879b1763f01dec55074ad4d72706eeb7d